groups_admin.rb

```diff
@@ -100,34 +125,180 @@ def update_accept_header(extension,mime_type)
 # groups/<group id> (PUT GET DELETE)
 
 ################ groups ###############
-
+# create group representations
+# input:  https://developers.google.com/admin-sdk/directory/v1/guides/manage-groups
+# output: https://developers.google.com/admin-sdk/directory/v1/reference/groups#resource
 
 # list groups
 ## Return list of ids or list of array/hash [id,url]
 ## put / post must accept (optional) settings
-get '/groups' do
-  "get all groups via group email"
+
+get '/groups/?', :provides => [:json, :html] do
+  respond_to do |format|
+    config = {
+        :args => nil,
+        :method_symbol => :list_groups,
+        :handle_result => Proc.new { |result| result },
+        :service_name => 'ADMIN_DIRECTORY'
```

Comment:
That's Google's name for their service.  This is a reference to the section of the yaml configuration file that has the required details for invoking the service.